src/modules/qt/filter_qresize.cpp

```diff
@@ -0,0 +1,344 @@
+		int real_width = mlt_properties_get_int( properties, "meta.media.width" );
+		int real_height = mlt_properties_get_int( properties, "meta.media.height" );
+		if ( real_width == 0 )
+			real_width = mlt_properties_get_int( properties, "width" );
+		if ( real_height == 0 )
+			real_height = mlt_properties_get_int( properties, "height" );
+		double input_ar = aspect_ratio * real_width / real_height;
+		double output_ar = consumer_aspect * owidth / oheight;
+		
+// 		fprintf( stderr, "real %dx%d normalised %dx%d output %dx%d sar %f in-dar %f out-dar %f\n",
+// 		real_width, real_height, normalised_width, normalised_height, owidth, oheight, aspect_ratio, input_ar, output_ar);
+
+		// Optimised for the input_ar > output_ar case (e.g. widescreen on standard)
+		int scaled_width = rint( ( input_ar * normalised_width ) / output_ar );
+		int scaled_height = normalised_height;
+
+		// Now ensure that our images fit in the output frame
+		if ( scaled_width > normalised_width )
+		{
+			scaled_width = normalised_width;
+			scaled_height = rint( ( output_ar * normalised_height ) / input_ar );
+		}
+
+		// Now calculate the actual image size that we want
+		owidth = rint( scaled_width * owidth / normalised_width );
+		oheight = rint( scaled_height * oheight / normalised_height );
+
+		// Tell frame we have conformed the aspect to the consumer
+		mlt_frame_set_aspect_ratio( frame, consumer_aspect );
+	}
+
+	mlt_properties_set_int( properties, "distort", 0 );
+
+	// Now pass on the calculations down the line
+	mlt_properties_set_int( properties, "resize_width", *width );
+	mlt_properties_set_int( properties, "resize_height", *height );
+
+	// If there will be padding, then we need packed image format.
+	if ( *format == mlt_image_yuv420p )
+	{
+		int iwidth = mlt_properties_get_int( properties, "width" );
+		int iheight = mlt_properties_get_int( properties, "height" );
+		if ( iwidth < owidth || iheight < oheight )
+			*format = mlt_image_yuv422;
+	}
+
+	// Now get the image
+	if ( *format == mlt_image_yuv422 )
+		owidth -= owidth % 2;
+	error = mlt_frame_get_image( frame, image, format, &owidth, &oheight, writable );
+
+	if ( error == 0 && *image && *format != mlt_image_yuv420p )
+	{
+		*image = frame_resize_image( frame, *width, *height, *format );
+	}
+
+	return error;
+}
+
+/** Filter processing.
+*/
+
+static mlt_frame filter_process( mlt_filter filter, mlt_frame frame )
+{
+	// Store the aspect ratio reported by the source
+	mlt_deque_push_back_double( MLT_FRAME_IMAGE_STACK( frame ), mlt_frame_get_aspect_ratio( frame ) );
+
+	// Push this on to the service stack
+	mlt_frame_push_service( frame, filter );
+
+	// Push the get_image method on to the stack
+	mlt_frame_push_get_image( frame, filter_get_image );
+
+	return frame;
+}
+
+extern "C" {
+
+/** Constructor for the filter.
+*/
+
+mlt_filter filter_qresize_init( mlt_profile profile, mlt_service_type type, const char *id, char *arg )
+{
+	mlt_filter filter = mlt_filter_new();
```

Comment:
If this filter will be compiled for all versions of QT, then a version check here could return null unless QT_VERSION >= 0x050200.